src/globus_sdk/_generate_init.py

```diff
@@ -136,6 +136,8 @@ def __getattr__(name: str) -> t.Any:
         (
             "ComputeClient",
             "ComputeAPIError",
+            "FunctionDocument",
+            "FunctionMetadata",
```

Comment:
I _think_ we should name these `ComputeFunctionDocument` and `ComputeFunctionMetadata`, although I could be convinced otherwise.

Right now, we have a very flat namespace for the service-related SDK exports -- `globus_sdk.X` for all `X`.
In most cases they're unambiguously associated with a given service, e.g., `StorageGatewayPolicies` is associated with GCS and couldn't have meaning with respect to anything else.

In cases where user confusion seems possible, we opt for longer names, e.g., `SearchScrollQuery` rather than `ScrollQuery`.

There's no hard-and-fast rule here, so there's nothing definitively _wrong_ with `FunctionDocument` and `FunctionMetadata` as names. The underlying problem we're looking at is that services haven't been given their own namespaces to populate freely -- and it's something I've been mulling for a few years without having any clear plan of action yet.

When I look at these names not only in light of the Globus ecosystem, but in comparison with the broader Python ecosystem, I think `FunctionMetadata` looks too much like the type of object one would expect from `inspect` or `types` in the stdlib -- and the SDK does contain some special values in the Python context, like `MISSING`, which make that kind of confusion possible.

Reply:
No objections from me. This will be more relevant for the endpoint methods because we need to differentiate between GCS and Compute endpoints.